### PR TITLE
Euclidean algorithms

### PR DESCRIPTION
**Pull request by 1711shashank**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

LCM and GCD in cpp

### Checklist

- [✔] I've read the contribution guidelines.
- [✔] I've checked the issue list before deciding what to submit.
- [✔] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1612

### CODE REVIEW

File: C-Plus-Plus/math/LCM_and_GCD.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
+//LCM and GCD between two numbers
+
+#include <bits/stdc++.h>
+using namespace std;
+
+int gcd(int num1, int num2)
```
Comment (atarax665) [inline C-Plus-Plus/math/LCM_and_GCD.cpp]: This is an implementation of ```Euclidean Algorithm```. Remove the ```lcm function``` and add another program to calculate LCM separately without using ```GCD```. Rename this file as ```Euclidean_Algorithm.cpp```.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Please remove all the programs related to recursion based problems and make a new PR for it as it resolves another individual issue.

File: C-Plus-Plus/math/Euclidean_algorithms.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/*Euclidean algorithms
```
Comment (atarax665) [inline C-Plus-Plus/math/Euclidean_algorithms.cpp]: Add a short description of Euclidean algorithm for GCD.

File: C-Plus-Plus/math/Euclidean_algorithms.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+/*
+Input:
+Enter the number : 5 7  
+
+Output:                                                                                                                                    
+GCD = 1 
+
+
+space complexity = O(1)
```
Comment (atarax665) [inline C-Plus-Plus/math/Euclidean_algorithms.cpp]:
```suggestion
Space Complexity = O(1)
```

File: C-Plus-Plus/math/Euclidean_algorithms.cpp [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+time complexity = O(log(n))
```
Comment (atarax665) [inline C-Plus-Plus/math/Euclidean_algorithms.cpp]:
```suggestion
Time Complexity = O(log(n))
```

**Review by atarax665 — CHANGES_REQUESTED** [review]
The code looks good to me, just do the below mentioned changes so that I can approve this PR.
- Remove all the unnecessary files.
- Rename the file name from ```Euclidean_algorithms.cpp``` to ```Euclidean_Algorithm.cpp```.
- Modify the README.md accordingly.

**Comment by HarshCasper** [comment]
Please don't push 28 Commits for 4 Code! Push your Commits into a single atomic commit and add a new line at the end of the file. If you don't know how to do that, mention me here and I will share a detailed walkthrough to make that possible.

**Comment by 1711shashank** [comment]
@HarshCasper  please share the detailed.

**Comment by HarshCasper** [comment]
> @HarshCasper please share the detailed.

Let's say: there were 5 commits in total:

- `git reset --soft HEAD~5` will soft reset the last 5 commits.
- Make a single commit with `git add` and `git commit`
- And after that, `git push -f origin HEAD` will force push the single commit.

**Comment by 1711shashank** [comment]
@HarshCasper 👍

**Comment by 1711shashank** [comment]
> > @HarshCasper please share the detailed.
> 
> Let's say: there were 5 commits in total:
> 
> * `git reset --soft HEAD~5` will soft reset the last 5 commits.
> * Make a single commit with `git add` and `git commit`
> * And after that, `git push -f origin HEAD` will force push the single commit.

image

@HarshCasper 

`git reset --soft HEAD~5`  after running this cmd its asking to update all this changes 
Should I add all these changes in the next commit ???

**Comment by HarshCasper** [comment]
> > > @HarshCasper please share the detailed.
> > 
> > 
> > Let's say: there were 5 commits in total:
> > 
> > * `git reset --soft HEAD~5` will soft reset the last 5 commits.
> > * Make a single commit with `git add` and `git commit`
> > * And after that, `git push -f origin HEAD` will force push the single commit.
> 
> image
> 
> @HarshCasper
> 
> `git reset --soft HEAD~5` after running this cmd its asking to update all this changes
> Should I add all these changes in the next commit ???

Oh gosh 🤦‍♂️ 

Please see my original comment. I said `Let's say`. That doesn't mean you had to use the same commands now. Your PR has 21 Commits so you should have change it aptly. The Correct Command should have been `git reset --soft HEAD~21`. Please take a look at how Git works properly.

**Comment by 1711shashank** [comment]
> git reset --soft HEAD~21

I have used this cmd only

**Comment by HarshCasper** [comment]
> > git reset --soft HEAD~21
> 
> I have used this cmd only

Now make a Commit 🚀

File: C-Plus-Plus/recursion/All_possible_subset_using_recursion.cpp [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+Input 
+abc
+
+Output
+a c b bc ac ab abc
+
+
+Space complexity => O(1) 
+Time complexity => O(2^n-1) if 'n' is the length of string
+*/
```
Comment (atarax665) [inline C-Plus-Plus/recursion/All_possible_subset_using_recursion.cpp]: Add a line break at the end of the program.

**Review by atarax665 — CHANGES_REQUESTED** [review]
@1711shashank Please remove the recursion based programs as they don't belong to this PR. This PR is almost ready to merge just resolve all the conversation ASAP.

File: C-Plus-Plus/recursion/All_possible_subset_using_recursion.cpp (hunk 5 above)
Comment (1711shashank) [reply C-Plus-Plus/recursion/All_possible_subset_using_recursion.cpp]: @atarax665 
line break means
Do you mean i need to add a line at the end (after line 43 I need to add one more line)

Comment (atarax665) [reply C-Plus-Plus/recursion/All_possible_subset_using_recursion.cpp]: Yeah

Comment (1711shashank) [reply C-Plus-Plus/recursion/All_possible_subset_using_recursion.cpp]: @atarax665  Done
 
Any more changes or its ready to merge ??

**Comment by HarshCasper** [comment]
I guess there are a lot of changes through a single Commit 🤦‍♂️ 

@atarax665 I guess it would be best to close this PR and let @1711shashank make another one.

**Comment by atarax665** [comment]
> I guess there are a lot of changes through a single Commit 
> 
> @atarax665 I guess it would be best to close this PR and let @1711shashank make another one.

Yeah, @1711shashank I am closing this PR, you can raise a new PR from a new branch with appropriate changes.

**Pull request closed without merging**